src/services/upload_repository.js

```diff
@@ -40,27 +38,35 @@ export default class UploadRepository {
   }
 
   async receiptWithResult(uploadReceiptPromise, uploadResult) {
-    const uploadReceipt = await uploadReceiptPromise;
-    return {
-      ...uploadReceipt,
-      uploadResult
-    };
+    try {
+      const uploadReceipt = await uploadReceiptPromise;
+      this.handleLowBalanceWarning(uploadReceipt);
+      return {
+        ...uploadReceipt,
+        uploadResult
+      };
+    } catch (error) {
+      if (error instanceof InsufficientFundsToUploadBundleError) {
+        this.sentry.captureException(error);
+        throw error;
+      }
```

Comment:
I'd move rethrow outside of if, cause if we will catch some other type of error, it will be ignored